CharlieBackend.Api/Middlewares/IsAccountActiveMiddleware.cs

```diff
@@ -59,32 +51,24 @@ public async Task InvokeAsync(HttpContext context, IAccountService accountServic
 
                     if ((bool)!isActive)
                     {
-                        context.Response.StatusCode = 401;
+                        context.Response.StatusCode = StatusCodes.Status403Forbidden;
 
-                        await context.Response.WriteAsync("Account is not active!");
+                        await context.Response.WriteAsync("Account is not active(i.e. you don't have proper role). You need to authorize!");
                     }
 
                     if (context.Request.Path.Value.Contains("lessons"))
                     {
-                        var role = tokenS.Claims
-                                .First(claim => claim.Type == ClaimsIdentity.DefaultRoleClaimType).Value;
+                        var role = currentUserService.Role;
 
-                        if (role == "2")
+                        if (role == UserRole.Mentor)
                         {
-                            var id = tokenS.Claims.First(claim => claim.Type == "Id").Value;
+                            var id = currentUserService.EntityId;
 
                             context.Items["mentorId"] = id;
                         }
```

Comment:
Research if this IF statement is used for something

Reply:
Yeah, you were right